File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cpp
#include <vector>

using namespace std;

const int MOD = 1'234'567;

int solution(int n) {
  vector<int> dp(n + 1);

  dp[0] = 0;
  dp[1] = 1;
  for (int i = 2; i <= n; i++) dp[i] = (dp[i - 2] + dp[i - 1]) % MOD;
  return dp[n];
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {

    private static final int MOD = 1_234_567;

    public int solution(int n) {
        if (n == 0) {
            return 0;
        }

        int prev = 0;
        int curr = 1;

        for (int i = 2; i <= n; i++) {
            int next = (prev + curr) % MOD;

            prev = curr;
            curr = next;
        }
        return curr;
    }
}